modules/wc-product-editor/assets/js/e-wc-product-editor.js

```diff
@@ -0,0 +1,63 @@
+import { WooHeaderItem } from '@woocommerce/admin-layout';
+import { useEntityId } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { registerPlugin } from '@wordpress/plugins';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+function EditWithElementorButton() {
+	const productId = useEntityId( 'postType', 'product' );
+	const { saveEntityRecord } = useDispatch( 'core' );
+
+	const postStatus = useSelect( ( select ) => {
+		return select( 'core' ).getEditedEntityRecord( 'postType', 'product', productId )?.status;
+	}, [ productId ] );
+
+	const handleClick = () => {
+		if ( 'auto-draft' === postStatus ) {
+			saveEntityRecord( 'postType', 'product', {
+				id: productId,
+				name: `Elementor #${ productId }`,
+				status: 'draft',
+			} ).then( () => {
+				redirectToElementor( productId );
+			} ).catch( () => {} );
+		} else {
+			redirectToElementor( productId );
+		}
+	};
+
+	const redirectToElementor = ( id ) => {
+		const checkSaveStatus = () => {
+			if ( wp.data.select( 'core/editor' ).isSavingPost() ) {
```

Comment:
Instead of creating a recursive timeout, you could grab this value outside of this block:

```js
const isSaving = wp.data.select( 'core/editor' ).isSavingPost();
```

Create a state for when you should be redirecting `isRedirecting` and set to `true` inside your `handleClick`.  Then add a `useEffect` to watch for those things:

```js
useEffect( () => {
    if ( isRedirecting && ! isSaving ) {
        // do redirect
    }
}, [ isRedirecting, isSaving ] );
```

This will be more performant and reduce any potential delays due to the timeout.

Reply:
Of course! Updated.